webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteMenu.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import groupBy from 'lodash/groupBy';
+import { TypeAheadSelect } from 'patternfly-react';
+import SubstringWrapper from '../../common/SubstringWrapper';
+
+const { Menu, MenuItem } = TypeAheadSelect;
+
+const AutoCompleteMenu = ({ results, menuProps }) => {
+  let itemIndex = 0;
```

Comment:
I don't think you actually need to manage you own index.
The `map` function can give you the current index together with the value.

```js
Object.keys(grouped).sort().map((category, index) => ... );
```

Reply:
The only problem is that on every new category,
the index will be reset to 0.
this means that there would be to children with the same key, 
and there would be an error. :(

Reply:
What about?
```jsx
<MenuItem key=`${category}-${resultIndex}` />
```

Reply:
Still doesn't work as we need to count the menuItems position,
e.g. : ` <MenuItem key={itemIndex} option={result.label} position={itemIndex}>`